eosclubhouse/quests/hack2/onboardtour.py

```diff
@@ -0,0 +1,352 @@
+
+        self.show_highlight_widget('panel')
+        self.wait_confirm('DESKTOP_PANEL')
+        if self.os_previous == 'win':
+            self.wait_confirm('DESKTOP_PANEL_WIN')
+        if self.os_previous == 'mac':
+            self.wait_confirm('DESKTOP_PANEL_MAC')
+        self.onboarding_clean()
+        return self.step_apps
+
+    def step_apps(self):
+        self.wait_confirm('APPS1')
+        self.wait_confirm('APPS2')
+
+        # Wait for user to open App Center
+        #   We need to do it this way because g-s is always running,
+        #   so we detect foregrounding instead
+        self.show_highlight_widget('App Center')
+        self.show_hints_message(message_id='APPS3')
+        self.wait_for_app_in_foreground(app=self.softwareApp)
+
+        # TODO: clippy highlight Search Box in App Center
+        #   self.softwareApp.highlight_object('search_bar')
+        self.wait_confirm('APPS4')
+
+        if self.os_previous == 'win':
+            self.wait_confirm('APPS4_WIN')
+        if self.os_previous == 'mac':
+            self.wait_confirm('APPS4_MAC')
+
+        if self.has_connection():
+            askinstall_action = self.show_choices_message('APPS_Q_HASNET',
+                                                          ('APPS_A_YES', None, True),
+                                                          ('APPS_A_NO', None, False)).wait()
+            if askinstall_action.future.result():
+                self.onboarding_clean()
+                return self.step_apps_install
+            else:
+                self.onboarding_clean()
+                self.wait_confirm('APPS_NOINSTALL')
+        else:
+            self.wait_confirm('APPS_NONET')
+        return self.step_apps_writer
+
+    def step_apps_install(self):
+        self.wait_confirm('APPS_INSTALL1')
+        # TODO: clippy highlight Search
+        self.wait_confirm('APPS_INSTALL2')
+        # TODO: can clippy highlight the download button?
+        self.wait_confirm('APPS_INSTALL3')
+        self.wait_confirm('APPS_INSTALL4')
+        self.wait_confirm('APPS_INSTALL5')
+        return self.step_apps_writer
+
+    def step_apps_writer(self):
+        # TODO: wait for LibreOffice open here
+        self.show_highlight_icon('org.libreoffice.LibreOffice.writer')
+        self.wait_confirm('OPENWRITER')
+        self.onboarding_clean()
+
+        # TODO: Ideally we use this...
+        # self.ask_for_app_launch(app=self.officeApp, message_id='OPENWRITER')
+        #       Or this...
+        # self.show_hints_message(message_id='OPENWRITER')
+        # self.wait_for_app_in_foreground(app=self.softwareApp)
+        #
+        # TODO: this works, but it blocks execution and precludes a dialog box,
+        #   so we can't use it
+        # max_attempts = 50
+        # wait_seconds = 0.3
+        # for t in range(max_attempts):
+        #     if Desktop.app_is_running('org.libreoffice.LibreOfficeIpc0'):
+        #         break
+        #     sleep(wait_seconds)
```

Comment:
Now you can do it! See https://phabricator.endlessm.com/T30904#858354

Please rebase your branch with master and use the `ask_for_app_launch`. Make sure to pass `is_gtk=False` when instantiating the app.

Reply:
All done, works quite well - with the exception of Chrome, but that's probably because of the 'fake flatpak' that's being resolved in a future release.